File: website/src/interfaces/Org.ts
import type { Domain } from "./Domain";
import type { Member } from "./Member";

export interface Organization {
    _id?: string;
    name: string;
    domains: Domain[];
    members: Member[];
}

export type OrganizationCreate = Omit<Organization, 'members'> & {
    members: string;
};

export interface OrganizationEdit extends Organization {
    uploadedMembers?: string;
}

File: flutter/lib/models/org.dart
import 'package:examduler/models/domain.dart';
import 'package:examduler/models/member.dart';

class Org {
  final String? id;
  final String name;
  final List<Domain> domains;
  final List<Member> members;

  Org({
    this.id,
    required this.name,
    required this.domains,
    required this.members,
  });
}
